conclave-enclave/src/main/kotlin/com/r3/conclave/enclave/internal/StreamEnclaveHostInterface.kt

```diff
@@ -0,0 +1,189 @@
+package com.r3.conclave.enclave.internal
+
+import com.r3.conclave.common.internal.*
+import com.r3.conclave.mail.internal.writeInt
+import com.r3.conclave.utilities.internal.getAllBytes
+import java.io.InputStream
+import java.io.OutputStream
+import java.nio.ByteBuffer
+import java.util.concurrent.ArrayBlockingQueue
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.Executors
+
+class StreamEnclaveHostInterface(
+        private val outputStream: OutputStream,     // Messages going to the host
+        private val inputStream: InputStream,       // Messages arriving from the host
```

Comment:
What's the reasoning for abstracting away the socket/port and working at the stream level? It seems whever we will use these classes we will need to write the same networking code. It's also not clear which side is the server and which side is the client.

I suggest simplifying and just deal directly with ports. Have both this class and its counterpart take in a port, and have the host side start up the socket server and the enclave connect to it (all on localhost). There's no need to have any retries at this stage. We can assume a stable localhost connection for now. (I think going forward for prod use we will use shared memory for the IPC).

Related to this, I would rename this class to `EnclaveToHostSocketInterface` or just `HostSocketInterface`. And in that vein `EnclaveHostInterface` should be renamed likewise.

Reply:
The idea was to keep the implementation agnostic of the stream implementation, so that we can use other stream implementations if we like (named pipes etc). The intention would be to add a `Socket(HostEnclave|EnclaveHost)Interface` class that is either a subclass of this one, or has an instance of this one as a member.

I also strongly disagree with the `EnclaveToHost` or `HostToEnclave` naming idea (marco suggested something similar) as these classes are bidirectional and handle incoming calls as well as outgoing ones. For reference `HostEnclaveInterface` is supposed to read as "the hosts interface to the enclave". If you can think of a name that better expresses this, I'm all for it!

Reply:
Drop "Enclave" from the name then. It makes the name more unwieldy and context should make it clear where the source is (since it's in the enclave mode). Then the KDoc can make explicity the source and target.

Reply:
Will leave as is for now due to large numbers of code changes required.